flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/GraphCsvReader.java

```diff
@@ -0,0 +1,471 @@
+ * the delimiters (row and field),  the fields that should be included or skipped, and other flags
+ * such as whether to skip the initial line as the header.
+ * The configuration is done using the functions provided in The {@link org.apache.flink.api.java.io.CsvReader} class.
+ */
+public class GraphCsvReader<K,VV,EV> {
+
+	private final Path vertexPath,edgePath;
+	private final ExecutionEnvironment executionContext;
+	protected CsvReader EdgeReader;
+	protected CsvReader VertexReader;
+	protected MapFunction<K, VV> mapper;
+
+//--------------------------------------------------------------------------------------------------------------------
+
+	public GraphCsvReader(Path vertexPath,Path edgePath, ExecutionEnvironment context) {
+		this.vertexPath = vertexPath;
+		this.edgePath = edgePath;
+		this.VertexReader = new CsvReader(vertexPath,context);
+		this.EdgeReader = new CsvReader(edgePath,context);
+		this.mapper=null;
+		this.executionContext=context;
+	}
+
+	public GraphCsvReader(Path edgePath, ExecutionEnvironment context) {
+		this.vertexPath = null;
+		this.edgePath = edgePath;
+		this.EdgeReader = new CsvReader(edgePath,context);
+		this.VertexReader = null;
+		this.mapper = null;
+		this.executionContext=context;
+	}
+
+	public GraphCsvReader(Path edgePath,final MapFunction<K, VV> mapper, ExecutionEnvironment context) {
+		this.vertexPath = null;
+		this.edgePath = edgePath;
+		this.EdgeReader = new CsvReader(edgePath,context);
+		this.VertexReader = null;
+		this.mapper = mapper;
+		this.executionContext=context;
+	}
+
+	public GraphCsvReader (String edgePath,ExecutionEnvironment context) {
+		this(new Path(Preconditions.checkNotNull(edgePath, "The file path may not be null.")), context);
+
+	}
+
+	public GraphCsvReader(String vertexPath, String edgePath, ExecutionEnvironment context) {
+		this(new Path(Preconditions.checkNotNull(vertexPath, "The file path may not be null.")),new Path(Preconditions.checkNotNull(edgePath, "The file path may not be null.")), context);
+	}
+
+
+	public GraphCsvReader (String edgePath, final MapFunction<K, VV> mapper, ExecutionEnvironment context) {
+			this(new Path(Preconditions.checkNotNull(edgePath, "The file path may not be null.")),mapper, context);
+	}
+
+	public CsvReader getEdgeReader() {
+		return this.EdgeReader;
+	}
+
+	public CsvReader getVertexReader() {
+		return this.VertexReader;
+	}
+	//--------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types
+	 *
+	 * This method is overloaded for the case in which Vertices don't have a value
+	 *
+	 * @param type0 The type of CSV field 0 and 1 for the CSV reader used for reading Edge data, the type of CSV field 0 for the CSV reader used for reading Vertex data  and the type of Vetex ID in the returned Graph.
```

Comment:
I think this description is quite confusing..
Is `type0` the key type, `type1` the vertex value type and `type2` the edge value type?